public/app/plugins/datasource/influxdb/components/VisualInfluxQLEditor/FromSection.tsx

```diff
@@ -52,6 +52,9 @@ export const FromSection = ({
         value={policy ?? 'using default policy'}
         loadOptions={handlePolicyLoadOptions}
         onChange={(v) => {
+          if (v === null) {
+            return; // we do not handle the clear-value case, it should not happen, we did not enable it
+          }
```

Comment:
isn't this check superfluous? is it possible for `v` to be null if the component is not clearable?

also applies to the other down in the same file and in `public/app/plugins/datasource/influxdb/components/VisualInfluxQLEditor/TagsSection.tsx`

Reply:
AFAIK Segments and Selects are generic, I think the problem comes because you are explicitly using `onChange: (v: SelVal | null) => void;`. If somehow you make `Seg`'s props generic, you should be able to use them like `<Seg<SelVal> ...` and `<Seg<SelVal | null>` depending on the context (or better, let TypeScript infer it for you).

what do you think?